### ISSUE TITLE
J2store upgrade to j2commerce

### ISSUE THREAD

**Issue by niaziblog** [j2commerce/j2cart#45]
I have been using the j2store for a long time
Can I install the j2commerce? 
Are products automatically replaced and orders?
There is no problem in the sales section?

J2store is a very good store
But I am very happy that now a strong team led its development
good luck

**Comment by newlinewebdesign** [comment]
@niaziblog Those are great questions and we will be updating the j2commerce.com website with more details. These updates will be on top of your current J2Store v4.x version so you can update your current joomla4/5 sites seamlessly. These updates are designed to improve the UI/UX for admins while fixing bugs and updating every file to ensure compatibility with Joomla 6.

**Comment by niaziblog** [comment]
Thanks @newlinewebdesign  for reply. 
In fact, those who have j2store shop should not worry at all about the deletion of the inventory and details of the product store.
Because j2Commerce is the advanced and rebranded version of j2store
right?

**Comment by newlinewebdesign** [comment]
That is correct. I myself have over 50 j2store sites so each update needs to be seamless so it doesn't cause any major inconvenience.

**Comment by antoniotheant** [comment]
Hello, is J2commerce ok for Joomla 5.x?

**Comment by newlinewebdesign** [comment]
@antoniotheant yes, it's been tested on both 5.2 and 4.4.9. If you already have j2store running on a Joomla 4 site, you can simply install this on top of it without issue. If you are still on Joomla 3, you will need to go through the migration component and follow the steps to updating to Joomla 4/5.

**Comment by antoniotheant** [comment]
Thanks a lot @newlinewebdesign .
Do you know when plugins such as Subscriptions and Memberships, Simple Subscription Product, Re-Order will also be available?

**Comment by newlinewebdesign** [comment]
We are working on getting the new apps section online this week and I know both have already been updated.

**Comment by hajold** [comment]
Hi, I have Problems.

0 implode(): Argument #1 ($array) must be of type array, string specified

appears in the backend when I save an article with the tab “J2Store Cart” in v4.05
in 4.07 the error is no longer present.

However in the frontend error:
0
Attempt to assign property “availability” on null

the website was migrated from J3 via J4 to J5 according to the migration instructions
J2Store has been in use here since 2016

Please look at the website with call stack
https://hans-helle.com/eifelverein/verlag/buecher-und-karten/buecher
also
https://hans-helle.com/eifelverein/verlag/buecher-und-karten

Deprecated: Using ${var} in strings is deprecated, use {$var} instead in /home/hanshell/public_html/eifelverein/libraries/f0f/less/less.php on line 2428

from 2027 to 2038

// Type based operators
$fname = "op_${ltype}_${rtype}";

if (is_callable(array($this, $fname)))
{
$out = $this->$fname($op, $left, $right);

if (!is_null($out))
{
return $out;
}
}
Can anyone help?

**Comment by j2commerce** [comment]
Hello @hajold, thank you for the feedback.

Note that for specific issues, in the future, it is best to open separate issues here on Github, or else they can be lost in long threads. Issues that are fixed can be closed.

Ok, there are a few different things that seem to be happening here:

1 - the use of F0F in later versions of PHP will produce messages about deprecated code. These are not errors and your site is not affected. Once your site runs as expected, you can turn off error reporting from the global configuration of the site. So that goes for all messages starting with 'Deprecated:'.

2 - thanks for reporting the issue 'Attempt to assign property "availability" on null'. We are looking into it.

**Comment by antoniotheant** [comment]
Hi, is there any news about Subscriptions and Memberships, Simple Subscription Product and Re-Order plugins?

**Comment by newlinewebdesign** [comment]
@antoniotheant if I didn't catch this in my email feed, I would have never seen your question.  For items like this, please use our new support system so we can make sure your questions don't get lost:
[]((https://www.j2commerce.com/support/technical-support))
